src/client/entrypoints/main/board_tools.tsx

```diff
@@ -0,0 +1,125 @@
+import {BoardUpdateForm, NewBoardForm} from '_client/board_tools/board_form';
+import {BoardSelector, idSelector} from '_client/board_tools/board_selector';
+import * as UiUtil from '_client/common/ui_util';
+import {GameBoard} from '_client/game_board/controller/game_board';
+import {BoardModel} from '_client/game_board/model/board_model';
+import {BoardClient} from '_client/game_board/remote/board_client';
+import {DropdownSelector} from '_client/common/ui_components/dropdown';
+
+import React from 'react';
+import {loadBoard} from '_client/entrypoints/main/game_board';
+import ReactDOM from 'react-dom';
+import {Hideable} from '_client/common/ui_components/hideable';
+
+const NEW_BOARD_BUTTON = 'createNewBoard';
+const BOARD_FORM_STUB = 'createNewBoardFormStub';
+
+const MAIN_BOARD_STUB = 'mainBoard';
+
+const ACTIVE_SELECTOR_STUB = 'activeSelectorStub';
+const EDIT_SELECTOR_STUB = 'editSelectorStub';
+const EDITING_AREA_STUB = 'editingAreaStub';
+
+const SIDE_PANEL = 'sidePanelContent';
+
+const boardEditRow: React.CSSProperties = {
+  display: 'flex',
+  flexDirection: 'row',
+};
+
+export class EditingArea extends React.Component {
+  render(): JSX.Element {
+    return (
+      <div>
+        <div style={boardEditRow}>
+          <div id="activeSelectorStub"></div>
+          <div className="divider"></div>
+          <div id="editSelectorStub"></div>
+          <div className="divider"></div>
+        </div>
+        <br />
+        <div style={boardEditRow}>
+          <button id="createNewBoard" type="button" className="btn btn-primary">
+            Create New
+          </button>
+        </div>
+
+        <div id="createNewBoardFormStub" style={{zIndex: 30}}></div>
+
+        <br />
+        <br />
+
+        <div id="editingAreaStub"></div>
+      </div>
+    );
+  }
+}
+
+class BoardSelectors {
+  constructor(
+    private readonly activeSelector: DropdownSelector<string>,
+    private readonly editSelector: DropdownSelector<string>
+  ) {}
+
+  add(id: string): void {
+    this.activeSelector.add(idSelector(id, false));
+    this.editSelector.add(idSelector(id, true));
+  }
+}
+
+export function setupEditorPanel(): Hideable {
+  const root = document.createElement('div');
+  ReactDOM.render(<EditingArea />, root);
```

Comment:
Explicit renders against the ReactDOM like this are typically a smell to me, unless you're rolling it into existing HTML. Are we using these because the parent doesn't have JSX where we could just display them?

Reply:
(this has been moved to editor.tsx)

The reason why it's being done like this is that I have no idea what I'm doing. Basically, I want this component to be hidden until the right tab is clicked, so I'm attaching it to a root element that's not attached to the document yet. 

Then, when I want to display it, I'm attaching that root element to the document (and when I want to hide it, I'm detaching it). If there are better ways to accomplish this, do let me know.

Reply:
So if all you want is conditional display, you can do something like the below. Basically your function returns HTML to render if we're in a state it's desired, otherwise it returns null. That way they function is self-contained and you don't have to worry about the overall document, the function itself controls everything. I'm not positive that makes sense.

const App = () => {
  const [isVisible, setIsVisible] = useState(true);

  const toggleFunction = () => {
    setIsVisible(!isVisible);
  };

  return (
    <View style={styles.container}>
      {isVisible ? <Text style={styles.text}>Hello World!</Text> : null}
      <Button title="Toggle Visibility" onPress={toggleFunction} />
    </View>
  );
};

(taken from: https://reactnativeforyou.com/how-to-conditionally-hide-and-show-a-component-in-react-native/)

Reply:
Done as far as I could - it definitely feels a bit more Reacty now.

The problem is that all my underlying pre-React components assume that elements are actually on the real Document, so I can't swap everything over like this. I think the plan is to check this in for now and then clean this up when the remaining components have been converted to React.